File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        def fibo(n):
            phi = 1.61803398874989484820
            phi2 = (phi - 1) * (-1)
            return int(((phi ** n) - (phi2 ** n))//2.236067977)
        if n <= 2:
            return n
        return fibo(n + 1)
            
        

File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
      int dp[50];

    Solution() {
        // Initialize all dp elements to -1
        for(int i = 0; i < 50; i++) {
            dp[i] = -1;
        }
    }
    int solve(int n){
        if(n<=0) return dp[0];
        if(n==1) return dp[1];
        if(n==2) return dp[2];
        if(dp[n]!= -1) return dp[n];
        
        return dp[n] = solve(n-1) + solve(n-2);
    }
    int climbStairs(int n) {
        dp[0]=0;
        dp[1]=1;
        dp[2]=2;
        int ans = solve(n);
        return ans;
    }
};